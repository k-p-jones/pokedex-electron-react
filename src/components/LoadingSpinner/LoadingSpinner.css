div.loading-spinner {
  text-align: center;
}

.loading-spinner p {
  font-size: 30px;
}

.loading-spinner img.ui.image {
  height: 100px;
  width: auto;
  margin: 0 auto;
  animation-name: spin;
  animation-duration: 4000ms;
  animation-iteration-count: infinite;
  animation-timing-function: linear;
}

@keyframes spin {
  from {transform:rotate(0deg);}
  to {transform:rotate(360deg);}
}
