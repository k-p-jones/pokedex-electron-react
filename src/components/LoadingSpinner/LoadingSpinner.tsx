import React from 'react';
import { Image } from 'semantic-ui-react';
import pokeball from '../../images/pokeball.svg';
import './LoadingSpinner.css';

const LoadingSpinner: React.FC = () => (
  <div className="loading-spinner">
    <Image src={pokeball} />
    <p>Loading...</p>
  </div>
)

export default LoadingSpinner;
