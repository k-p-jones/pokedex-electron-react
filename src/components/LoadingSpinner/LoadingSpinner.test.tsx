import React from 'react';
import { render } from '@testing-library/react';
import LoadingSpinner from './LoadingSpinner';

test('it renders correctly', () => {
  const { container } = render(<LoadingSpinner />);
  expect(container).toMatchSnapshot();
});
