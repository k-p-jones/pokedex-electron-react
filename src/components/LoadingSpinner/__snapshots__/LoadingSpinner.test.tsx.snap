// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
<div>
  <div
    class="loading-spinner"
  >
    <img
      class="ui image"
      src="pokeball.svg"
    />
    <p>
      Loading...
    </p>
  </div>
</div>
`;
